src/psyclone/domain/lfric/lfric_stencils.py

```diff
@@ -0,0 +1,649 @@
+
+        :returns: a valid unique map name for a stencil in the PSy layer.
+        :rtype: str
+        '''
+        root_name = arg.name + "_stencil_map"
+        unique = LFRicStencils.stencil_unique_str(arg, "map")
+        return self._symbol_table.find_or_create_tag(unique, root_name).name
+
+    @staticmethod
+    def dofmap_symbol(symtab, arg):
+        '''
+        Creates and registers a symbol for the stencil dofmap associated with
+        the supplied kernel argument.
+
+        :param symtab: symbol table that will contain (or already contains) \
+            the symbol with this name.
+        :type symtab: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param arg: kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a dofmap symbol for a stencil in the PSy layer.
+        :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
+
+        '''
+        root_name = arg.name + "_stencil_dofmap"
+        unique = LFRicStencils.stencil_unique_str(arg, "dofmap")
+        if arg.descriptor.stencil['type'] == "cross2d":
+            num_dimensions = 4
+        else:
+            num_dimensions = 3
+        return symtab.find_or_create_array(root_name, num_dimensions,
+                                           ScalarType.Intrinsic.INTEGER,
+                                           tag=unique)
+
+    @staticmethod
+    def dofmap_size_symbol(symtab, arg):
+        '''
+        Create a valid symbol for the size (in cells) of a stencil
+        dofmap in the PSy layer.
+
+        :param symtab: symbol table that will contain (or already contains) \
+            the symbol with this name.
+        :type symtab: :py:class:`psyclone.psyir.symbols.SymbolTable`
+        :param arg: the kernel argument with which the stencil is associated.
+        :type arg: :py:class:`psyclone.dynamo0p3.DynKernelArgument`
+
+        :returns: a symbol for the stencil size.
+        :rtype: :py:class:`psyclone.psyir.symbols.Symbol`
+
+        '''
+        root_name = arg.name + "_stencil_size"
+        unique = LFRicStencils.stencil_unique_str(arg, "size")
+        if arg.descriptor.stencil['type'] == "cross2d":
+            num_dimensions = 2
+        else:
+            num_dimensions = 1
+        return symtab.find_or_create_array(root_name, num_dimensions,
+                                           ScalarType.Intrinsic.INTEGER,
+                                           tag=unique)
+
+    @staticmethod
+    def max_branch_length_name(symtab, arg):
+        '''
+        Create a valid unique name for the maximum length of a stencil branch
+        (in cells) of a 2D stencil dofmap in the PSy layer. This is required
+        in the kernels for defining the maximum possible length of one of the
+        dofmap array dimensions.
+
+        :param symtab: symbol table that will contain (or already contains) \
```

Comment:
Please remove the `\` character and align "the symbol" in the line below with "symbol table".